lib/content/utils.js

```diff
@@ -69,6 +69,118 @@ const check = {
   },
 };
 
+/**
+ * Grab type from all content types in system
+ *
+ * @param {object} req - Express Request Object
+ *
+ * @returns {promise} reject with error message; resolve with single content-type object
+ */
+const type = (req) => {
+  return new Promise((res, rej) => {
+    const typ = req.params.type.toString();
+
+    if (!typ || typ === '') {
+      rej(config.content.messages.missing.type.replace('%type', '(no type given)'));
+    }
+
+    const exists = req.content.types.find(tp => {
+      return tp.id === typ.toLowerCase();
+    });
+
+    if (!exists || (exists === false) || (typeof exists !== 'object') || !exists.id || !exists.name) {
+      rej(config.content.messages.missing.type.replace('%type', typ));
+    }
+
+    res(exists);
+  });
+};
+
+/**
+ * Determine if an id is in uuid format
+ *
+ * @param {object} req - Express Request Object
+ *
+ * @returns {promise} reject with error message; resolve true
+ */
+const id = (req) => {
+  return new Promise((res, rej) => {
+    // no id in the request
+    if (!req.params.hasOwnProperty('id')) {
+      res(true);
+    }
+
+    // this can be the add, approve, or save action instead of a uuid
+    if (req.params.id === config.content.actions.add || req.params.id === config.content.actions.save || req.params.id === config.content.actions.approve) {
```

Comment:
When would the `id` parameter (that is dynamic) be one of our hardcoded path items? Also, these are "true" for UUIDs?